File: report-service/src/main/java/com/example/report_service/dtos/InventoryMovementDto.java
package com.example.report_service.dtos;

import com.example.report_service.enums.TypeMovement;
import lombok.Data;

import java.util.List;

@Data
public class InventoryMovementDto {

    private List<ItemMovementDetailDto> movementDetail;
    private TypeMovement typeMovement;
    private Long userId;
    private Long reportId;
    private String reason;
}

File: Frontend/gestion-incidentes/src/app/dtos/inventory-movement.dto.ts
export interface ItemMovementDetailDto {
  resourceId: number;
  quantity: number;
}

export type TypeMovement = 'ENTRADA' | 'SALIDA';

export interface InventoryMovementDto {
  movementDetail: ItemMovementDetailDto[];
  typeMovement: TypeMovement;
  userId: number;
  reportId: number | null;
  reason: string;
}
